src/vm/checker/read_only/mod.rs

```diff
@@ -126,9 +141,26 @@ impl <'a, 'b> ReadOnlyChecker <'a, 'b> {
             Add | Subtract | Divide | Multiply | CmpGeq | CmpLeq | CmpLess | CmpGreater |
             Modulo | Power | BitwiseXOR | And | Or | Not | Hash160 | Sha256 | Keccak256 | Equals | If |
             ConsOkay | ConsError | DefaultTo | Expects | ExpectsErr | IsOkay | IsNone |
-            ListCons | GetBlockInfo | FetchEntry | FetchContractEntry | TupleGet | Print | AsContract | Begin => {
+            ListCons | GetBlockInfo | TupleGet | Print | AsContract | Begin => {
                 self.are_all_read_only(true, args)
             },
+            FetchEntry | FetchContractEntry => {
+                use vm::functions::tuples;
+                use vm::functions::tuples::TupleDefinitionType::{Implicit, Explicit};
+                
+                let res = match tuples::tuple_definition_type(&args[1]) {
+                    Implicit(ref tuple_expr) => {
+                        self.are_tuple_values_read_only(tuple_expr)
+                    },
+                    Explicit => {
```

Comment:
I don't think this is quite right ---

it should be:

```
Explicit => {
   self.are_all_read_only(true, args)
}
```

Otherwise, it will fail while checking something like:

```
(define (get-tuple) (tuple (account 0)))
(fetch-entry! tokens (get-tuple))
```

Because it's trying to _always_ parse that explicit argument as a `(tuple ...)` invocation.